telegram/ext/conversationhandler.py

```diff
@@ -592,6 +592,10 @@ def update_state(self, new_state: object, key: Tuple[int, ...]) -> None:
                     )
 
         elif new_state is not None:
+            if new_state not in self.states:
+                warnings.warn(
+                    "Handler returned state which is unknown to the ConversationHandler."
```

Comment:
Just an idea: If the hanlder has a name, we could include that in the message. Makes debugging easier :) And maybe including the unknown state is useful, too …